src/Streamly/Internal/FS/Event/Windows.hs

```diff
@@ -496,30 +488,26 @@ watchRecursive = watchWith (setRecursiveMode True)
 --
 -- /Pre-release/
 --
-watch :: NonEmpty (Array Word8) -> Stream IO Event
+watch :: NonEmpty Path -> Stream IO Event
 watch = watchWith id
 
 getFlag :: DWORD -> Event -> Bool
 getFlag mask Event{..} = eventFlags == mask
 
--- XXX Change the type to Array Word8 to make it compatible with other APIs.
---
 -- | Get the file system object path for which the event is generated, relative
--- to the watched root. The path is a UTF-8 encoded array of bytes.
+-- to the watched root.
 --
 -- /Pre-release/
 --
-getRelPath :: Event -> Array Word8
-getRelPath Event{..} = (UTF8.toArray . UTF8.pack) eventRelPath
+getRelPath :: Event -> Path
+getRelPath Event{..} = unsafePerformIO $ Path.fromString eventRelPath
```

Comment:
Do not use IO monad here. Make the return type `Maybe` and remove unsafePerformIO. The monad should be Maybe.  If it is guaranteed to be `Just` then we can use `fromJust`.